common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt

```diff
@@ -334,7 +328,7 @@ class WebSocketRepositoryImpl @Inject constructor(
             connected.complete(true)
         } else {
             connectionState = WebSocketState.CLOSED_AUTH
-            connected.completeExceptionally(AuthorizationException())
+            connected.complete(false)
```

Comment:
This will cause problems for future attempts to send a message to the websocket because of the check at the start of the `connect()` function: the completable is now completed succesfully, and the app will assume there is a connection it can use.
`completeExceptionally` shouldn't _throw_ an exception, it just cancels the job, so I don't think this change is necessary. If you'd still like to change it, the check at the start of `connect()` should be updated to only return true if the job is completed, not cancelled, and the completion result was `true`.

Reply:
Fixed